### PR TITLE
configure travis to check smoke and functional tests

### PR DESCRIPTION
**Pull request by henzigo**
| Q             | A
| ------------- | ---
|Description, reason for the PR|  With this PR, contributors will have better feedback, because smoke and functional tests are checked.
|New feature| Yes 
|[BC breaks](https://github.com/shopsys/shopsys/blob/master/docs/contributing/backward-compatibility-promise.md)| No 
|Fixes issues| ... 
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

**Comment by henzigo** [comment]
Targeted branch is 8.0 because of https://github.com/shopsys/shopsys/pull/933.

File: build.xml [hunk 1]
```diff
@@ -309,4 +309,6 @@
         </exec>
     </target>
 
+    <target name="travis-tests-functional" depends="composer-dev,db-create,db-import-basic-structure,db-migrations,create-domains-data,db-fixtures-demo,load-plugin-demo-data,generate-friendly-urls,test-db-create,grunt,error-pages-generate,test-db-demo,product-search-recreate-structure,product-search-export-products,tests-smoke,tests-functional" description="Run functional and smoke tests."/>
```
Comment (boris-brtan) [inline build.xml]: do we need db-create here? or do we need the whole travis-tests-functional phing target ?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
+dist: xenial
+sudo: required
+
+stages:
+  - Build
+  - Tests
+
+cache:
+  timeout: 1000
```
Comment (boris-brtan) [inline .travis.yml]: is 1000 enough?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
+dist: xenial
+sudo: required
```
Comment (boris-brtan) [inline .travis.yml]: isn't this default value?

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    - $CACHE_DIR
+
+env:
+  global:
+    - CACHE_DIR=$HOME/.cache/docker
+    - CACHE_FILE_IMAGE=$CACHE_DIR/php-fpm.tar.gz
+    - ES_VERSION=6.3.2
+
+before_install:
+  - if [ -f ${CACHE_FILE_IMAGE} ]; then gunzip -c ${CACHE_FILE_IMAGE} | docker load; fi
```
Comment (boris-brtan) [inline .travis.yml]:
```suggestion
  - if [ -f ${CACHE_FILE_IMAGE} ]; then docker load < ${CACHE_FILE_IMAGE}; fi
```
what about this way ?

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  - if [ $TRAVIS_BUILD_STAGE_NAME = Tests ]; then sudo apt-get update; fi
```
Comment (boris-brtan) [inline .travis.yml]: what if we use here multiline syntax ?
```
- |
  if [ $TRAVIS_BUILD_STAGE_NAME = Tests ]; then
    sudo apt-get update
    sudo apt-get install language-pack-cs language-pack-hu language-pack-pl language-pack-sk
    ......
  fi
```

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+cache:
+  timeout: 1000
+  directories:
+    - $CACHE_DIR
+
+env:
+  global:
+    - CACHE_DIR=$HOME/.cache/docker
+    - CACHE_FILE_IMAGE=$CACHE_DIR/php-fpm.tar.gz
+    - ES_VERSION=6.3.2
```
Comment (boris-brtan) [inline .travis.yml]: maybe we could put env variable for versions of services like ELASTICSEARCH, REDIS, POSTGRES into travis UI like we have it with COMPOSER_MEMORY_LIMIT

File: .travis.yml [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  - if [ -f ${CACHE_FILE_IMAGE} ]; then gunzip -c ${CACHE_FILE_IMAGE} | docker load; fi
+  - if [ $TRAVIS_BUILD_STAGE_NAME = Tests ]; then sudo apt-get update; fi
+  - if [ $TRAVIS_BUILD_STAGE_NAME = Tests ]; then sudo apt-get install language-pack-cs language-pack-hu language-pack-pl language-pack-sk; fi
+  - if [ $TRAVIS_BUILD_STAGE_NAME = Tests ]; then sudo /etc/init.d/postgresql stop; fi
+  - if [ $TRAVIS_BUILD_STAGE_NAME = Tests ]; then sudo /etc/init.d/postgresql start 10; fi
+  - if [ $TRAVIS_BUILD_STAGE_NAME = Tests ]; then curl -O https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-$ES_VERSION.deb && sudo dpkg -i --force-confnew elasticsearch-$ES_VERSION.deb && sudo service elasticsearch start; fi
+  - if [ $TRAVIS_BUILD_STAGE_NAME = Tests ]; then psql -c "CREATE USER root WITH SUPERUSER PASSWORD 'root';" -U postgres; fi
+
+addons:
+  postgresql: "10"
```
Comment (boris-brtan) [inline .travis.yml]: what about to have it like ENV_VAR ???

File: .travis.yml [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+  - redis-server
+  - docker
+
+jobs:
+  include:
+    - stage: Build
+      before_script:
+        - mkdir -p $CACHE_DIR
+      script:
+        - docker image build
```
Comment (boris-brtan) [inline .travis.yml]: current monorepo `.travis.yml` checks php 7.1 but this change will check php that is in the Dockerfile that is 7.3 by default. Is this change OK ?

File: .travis.yml [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+    - CACHE_FILE_IMAGE=$CACHE_DIR/php-fpm.tar.gz
+    - ES_VERSION=6.3.2
+
+before_install:
+  - if [ -f ${CACHE_FILE_IMAGE} ]; then gunzip -c ${CACHE_FILE_IMAGE} | docker load; fi
+  - if [ $TRAVIS_BUILD_STAGE_NAME = Tests ]; then sudo apt-get update; fi
+  - if [ $TRAVIS_BUILD_STAGE_NAME = Tests ]; then sudo apt-get install language-pack-cs language-pack-hu language-pack-pl language-pack-sk; fi
+  - if [ $TRAVIS_BUILD_STAGE_NAME = Tests ]; then sudo /etc/init.d/postgresql stop; fi
+  - if [ $TRAVIS_BUILD_STAGE_NAME = Tests ]; then sudo /etc/init.d/postgresql start 10; fi
+  - if [ $TRAVIS_BUILD_STAGE_NAME = Tests ]; then curl -O https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-$ES_VERSION.deb && sudo dpkg -i --force-confnew elasticsearch-$ES_VERSION.deb && sudo service elasticsearch start; fi
```
Comment (boris-brtan) [inline .travis.yml]: wouldn't it be better if we cache deb file?

File: .travis.yml [hunk 10]
```diff
@@ -0,0 +1,60 @@
+dist: xenial
```
Comment (boris-brtan) [inline .travis.yml]: xenial is 16.04
there is also bionic 18.04

**Comment by PetrHeinz** [comment]
A separate PR #1031 was created that tries to resolve the same thing.

**Comment by boris-brtan** [comment]
but with no BC break

**Pull request closed without merging**